### ISSUE TITLE
Sometimes error messages point to wildly wrong places

### ISSUE THREAD

**Issue by Hixie** [dart-lang/dartdoc#1405]
In routes.dart, we have a class ModalRoute, with a member willPop that has a dartdoc that dubiously refers to the bogus, maybe doubly-bogus, `[super.willPop()]`. That class extends a subclass of Route and also mixes in a class that extends Route directly. Route and the mixin both have a willPop. The mixin has no docs in its. The Route class does. Route is in navigator.dart. I get this error:

```
warning: unresolved doc reference [super.willPop()], from material.Route.willPop (/usr/local/google/home/ianh/dev/flutter/packages/flutter/lib/src/widgets/navigator.dart:76:31)
```

Best guess is that when you have an error in the doc of an override, dartdoc always points to the original member (in the superclass).

**Comment by jcollins-g** [comment]
Yep.  This is actually a bit more than just the warning message being wrong: in some cases, we're attempting to resolve names using the wrong context and then warning about it (this is most noticeable for parameters which can change their names when overridden).
